### PR TITLE
Parsing GenBank TLS

### PR DESCRIPTION
**Pull request by Andrey-Raspopov**
This pull request addresses issue #2620 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_GenBank.py [hunk 1]
```diff
@@ -2455,6 +2455,43 @@ def test_record_parser_tsa(self):
...
+                    ("/country=", '"Brazil: Manaus"'),
+                    ("/lat_lon=", '"2.92 S 59.95 W"'),
+                    ("/collection_date=", '"2013"'),
+                    ("/note=", '"metagenomic"'),
+                ),
+            )
+        ]
+        self.perform_record_parser_test(
+            record, length, locus, definition, accession, titles, features
+        )
```
Comment (peterjc) [inline Tests/test_GenBank.py]: Can you explicitly test the TLS line was read in?

(Might we worth adding something similar to the TSA test)

File: NEWS.rst [hunk 2]
```diff
@@ -47,6 +47,7 @@ possible, especially the following contributors:
 - Rob Miller
 - Sergio Valqui
 - Sujan Dulal (first contribution)
+- Andrey Raspopov
```
Comment (peterjc) [inline NEWS.rst]: You get to go first (alphabetical by first name).

File: Tests/GenBank/tls_KDHP01000000.gb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                     /isolation_source="soil samples in a lowland tropical
+                     evergreen rain forest in Amazonia"
+                     /db_xref="taxon:410658"
+                     /environmental_sample
+                     /country="Brazil: Manaus"
+                     /lat_lon="2.92 S 59.95 W"
+                     /collection_date="2013"
+                     /note="metagenomic"
+TLS         KBUV01000001-KBUV01003714
+//
```
Comment (peterjc) [inline Tests/GenBank/tls_KDHP01000000.gb]: Is that exactly as the NCBI did it - missing a new line character after the slash slash?

Comment (Andrey-Raspopov) [reply Tests/GenBank/tls_KDHP01000000.gb]: There is no new line here: https://www.ncbi.nlm.nih.gov/nuccore/GAAA00000000.1
Should we add it?

Comment (peterjc) [reply Tests/GenBank/tls_KDHP01000000.gb]: https://www.ncbi.nlm.nih.gov/nuccore/GAAA00000000.1 is HTML so no real guide.

Using the send to file menu, I get a trailing new line (as expected) and a blank line:

```
$ hexdump -C  ~/Downloads/sequence.gb  | tail
000009b0  76 69 64 65 64 20 62 79  20 44 72 2e 0a 20 20 20  |vided by Dr..   |
000009c0  20 20 20 20 20 20 20 20  20 20 20 20 20 20 20 20  |                |
000009d0  20 20 43 68 72 69 73 20  41 6d 65 6d 69 79 61 20  |  Chris Amemiya |
000009e0  6f 66 20 42 65 6e 61 72  6f 79 61 20 52 65 73 65  |of Benaroya Rese|
000009f0  61 72 63 68 20 49 6e 73  74 69 74 75 74 65 2c 20  |arch Institute, |
00000a00  53 65 61 74 74 6c 65 22  0a 54 53 41 20 20 20 20  |Seattle".TSA    |
00000a10  20 20 20 20 20 47 41 41  41 30 31 30 30 30 30 30  |     GAAA0100000|
00000a20  31 2d 47 41 41 41 30 31  30 39 33 35 30 37 0a 2f  |1-GAAA01093507./|
00000a30  2f 0a 0a                                          |/..|
00000a33

```

File: Tests/test_GenBank.py (hunk 1 above)
Comment (Andrey-Raspopov) [reply Tests/test_GenBank.py]: Can you please elaborate on that? The only mark that this is TLS is "TLS: " in the definition at the moment.

Comment (Andrey-Raspopov) [reply Tests/test_GenBank.py]: Also, there inside this method we are asserting length vs len(record.sequence), which is 0 for TSA and TLS. On the other hand we could use record.size for the same assertion.

Comment (peterjc) [reply Tests/test_GenBank.py]: TSA and TLS have no sequence and thus no meaningful length. They do have a record count.

I mean can the test confirm ``KBUV01000001-KBUV01003714`` (from the TLS line) is available within the parsed object.

Comment (Andrey-Raspopov) [reply Tests/test_GenBank.py]: Yes, got it, but one small detail. In current realisation we get list ['KBUV01000001', 'KBUV01003714'], and the same for TSA, shouldn't it be from the first to the last index?

Comment (peterjc) [reply Tests/test_GenBank.py]: I think you are suggesting expanding this to a list of all 3714 entries or something similar (like the Python 3 range function). I was sure there was an open issue on this idea, but I can't find it right now.... context could have been broader than just GenBank / EMBL records.

Comment (Andrey-Raspopov) [reply Tests/test_GenBank.py]: I've found this comment from you, that concerns the choice between widespread formats and language structures: https://github.com/biopython/biopython/issues/2131#issuecomment-502616697
The issue is indeed broader than this fix, so for now I'm settling on the latest update to the tests and awaiting reviews.

**Comment by peterjc** [comment]
Thank you!

**Pull request merged**